### DISCUSSION TITLE
"ownerReferences" consideration while deleting objects.

### DISCUSSION THREAD

**Discussion by sayan-biswas** [tektoncd/results#648] [General]
In the present implementation of the reconciler, we are checking the `ownerReferences` while deleting an object from the cluster. 

https://github.com/tektoncd/results/blob/6e51562c62f1f807e149bdc6a4ff5ee7b0ef33e5/pkg/watcher/reconciler/dynamic/dynamic.go#L185-L187

Added from #255 

While the intention specifies that those objects with `ownerReferences` are to be managed by the owner itself, but is there any value in keeping the object in the cluster even after it executed successfully and stored in results?
There are cases where other controllers will add `ownerReferences` for it purpose without knowing that TektonResults will ignore them, and they'll live in the cluster until deleted manually.

I suggest, if we need to keep this validation, we can keep it by default, and allow this behaviour to be altered with config parameters.

**Comment by adambkaplan** [comment]
I think this validation makes sense, period. We are ostensibly watching `PipelineRun` and `TaskRun` objects, which means we should get a reconcile event when a respective owner ref is updated/changed. I see this primarily impacting `PipelineRun` objects which have additional owner refs. If a `PipelineRun` is owned by another object, we IMO we should not prune it and wait for either:

- The owner to remove its owner ref, or
- The owner object to be deleted

Calling delete on an object will [cascade deletion](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#cascading-deletion) to its child objects by default, so Results shouldn't need to do anything regarding cleanup. And if the thing that deletes an object uses the `--cascade=orphan` option, Results should respect that.

**Comment by Roming22** [comment]
The issue is that we then have completed `PipelineRuns` hanging around because they have an ownerReference.

I don't see the value with keeping the `PipelineRun` resource on the cluster once all tasks have executed. Users should access the resources through Tekton-Results.

Also, I don't think the Tekton Pipelines pruner care about ownerReferences when pruning, so we have a change of behavior when going from the default pruner to Tekton-Results.

I'd rather there be an option to skip resources with ownerReferences that is set to `false` by default.

**Comment by Roming22** [comment]
There seems to be an agreement in the Working Group call to:

- Add a flag to enable/disable the behavior
- Have the default flag value disable the behavior, so Tekton Results behaves like the Tekton Pipelines pruner.
